main.go

```diff
@@ -5,75 +5,137 @@ import (
 )
 
 func main() {
-
 	flags := getCLflags()
-
 	conf := NewConfig()
-	log.Debug(conf.s3)
-
-	sb, err := newS3Backend(conf.s3)
-	if err != nil {
-		log.Fatal("Could not connect to s3 backend: ", err)
-	}
 
-	log.Infof("Connection to s3 bucket %v established", sb.Bucket)
-
-	var elastic *esClient
-	if conf.elastic != (elasticConfig{}) {
-		elastic, err = newElasticClient(conf.elastic)
+	switch flags.action {
+	case "es_backup":
+		elastic, err := newElasticClient(conf.elastic)
 		if err != nil {
 			log.Fatal(err)
 		}
-	}
-
-	var mongo mongoConfig
-	if conf.mongo != (mongoConfig{}) {
-		mongo = conf.mongo
-	}
-
-	var pg DBConf
-	if conf.db != (DBConf{}) {
-		pg = conf.db
-	}
-
-	switch flags.action {
-	case "es_backup":
-		err := elastic.backupDocuments(sb, conf.publicKeyPath, flags.name)
+		sb, err := newS3Backend(conf.s3)
 		if err != nil {
+			log.Fatal("Could not connect to s3 backend: ", err)
+		}
+
+		if err := elastic.backupDocuments(sb, conf.publicKeyPath, flags.name); err != nil {
 			log.Fatal(err)
 		}
 	case "es_restore":
-		err := elastic.restoreDocuments(sb, conf.privateKeyPath, flags.name, conf.c4ghPassword)
+		elastic, err := newElasticClient(conf.elastic)
+		if err != nil {
+			log.Fatal(err)
+		}
+		sb, err := newS3Backend(conf.s3)
 		if err != nil {
+			log.Fatal("Could not connect to s3 backend: ", err)
+		}
+
+		if err := elastic.restoreDocuments(sb, conf.privateKeyPath, flags.name, conf.c4ghPassword); err != nil {
 			log.Fatal(err)
 		}
 	case "mongo_dump":
-		err := mongo.dump(*sb, conf.publicKeyPath, flags.name)
+		mongo := conf.mongo
+		sb, err := newS3Backend(conf.s3)
 		if err != nil {
+			log.Fatal("Could not connect to s3 backend: ", err)
+		}
+
+		if err := mongo.dump(*sb, conf.publicKeyPath, flags.name); err != nil {
 			log.Fatal(err)
 		}
 	case "mongo_restore":
-		err := mongo.restore(*sb, conf.privateKeyPath, flags.name, conf.c4ghPassword)
+		mongo := conf.mongo
+		sb, err := newS3Backend(conf.s3)
 		if err != nil {
+			log.Fatal("Could not connect to s3 backend: ", err)
+		}
+
+		if err := mongo.restore(*sb, conf.privateKeyPath, flags.name, conf.c4ghPassword); err != nil {
 			log.Fatal(err)
 		}
 	case "pg_dump":
-		err := pg.dump(*sb, conf.publicKeyPath)
+		pg := conf.db
```

Comment:
Although not closely related to this PR, might be good to add the requirement of the tool `pg_dump` in the README.